### PR TITLE
PDS Mock Publisher 284

### LINKED ISSUE #284: Local PDS Simulator
This will probably be accomplished by creating a bunch of rospy python scripts.
We need to have some "fake" publishers to mock the real data that would be published if the rover actually on.

launch parameters : mode ...

- [x] Write initial publisher rospy script(s)
- - [ ] Temperature
- - - [ ] Stable
- - - [ ] Unstable
- - - - [ ] start stable and go past max
- - - - [ ] start stable and go past min
- - [x] Voltage

### PR DESCRIPTION
**Pull request by adrientremblay**
# Assignee Section

## Description
Creates a a PDS mock publisher node to publish mock values to the battery_voltage and battery_temps topics while in local mode.  The publisher defaults to stable mode for everything but the mode for the voltage and for each of the three temperatures can be changed using the command `rosparam set /<param> <mode>`.  Where <param is the ROS parameter, and <mode> is the mode string.

The Parameters are as follows:

- PDS_mock_mode_voltage : voltage mode
- PDS_mock_mode_temp1 : temperature 1 mode
- PDS_mock_mode_temp2 : temperature 2 mode
- PDS_mock_mode_temp3 : temperature 3 mode

The modes are as follows:

- stable
- rise
- fall

Modes work by shifting towards a target value.  Therefore changing the mode simply just changes said target value.  

So far the target values are as follows:

**Voltage**
max : 20
min : 10
safe : 15

**Temperature**
max : 100
min : -20
safe : 50

closes #284 
The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.

- [ ] Local Test Performed Successfully
- [ ] [General Integration Test](https://docs.google.com/document/d/1ug0CpA1cIzURP8DDFSvCt2CEJJSwJ6Ta6B1LG_hYk6I/edit) Performed Successfully

##Local Testing Guide

1. navigate to ~/Programming/robotics-prototype/robot/util/rosRoverStart and run `roslaunch local_rover.launch `
2. A voltage of 15V and tempeatures of 50C (for all three temp indicators) should appear on the basestation
3. Test every mode using `rosparam set /<param> <mode>`
Ex: `rosparam set PDS_mock_mode_temp3 rise` would set T3 to the rise mode which drives it's temperature up to 100C

### CODE REVIEW

File: robot/util/rosRoverStart/local_rover.launch [hunk 1]
```diff
@@ -2,4 +2,5 @@
     <node pkg="mux_selector" type="mux_select_server.py" name="mux_select_server" output="screen" args="local"/>
     <node pkg="web_video_server" type="web_video_server" name="web_video_server" output="screen"/>    
     <node pkg="task_handler" type="task_handler_server.py" name="task_handler_server" output="screen" args="local"/>
+    <node pkg="mcu_control" type="PDSMockPublisher.py" name="PDS_mock_pub" output="screen"/>
```
Comment (MartensCedric) [inline robot/util/rosRoverStart/local_rover.launch]: `PDSLocalPublisher`?

**Comment by adrientremblay** [comment]
Cedrics' big brain google hangout pseudocode :+1: 

Cedric Martens4:12 PM
PDS_mock_mode_voltage
PDS_voltage_mode = 'stable'
PDS_voltage_value = 15
Cedric Martens4:13 PM
PDS_voltage_mode='rise'
PDS_voltage_value = 20
PDS_voltage_rate = 0.1V
PDS_voltage_noise = 0.3
Cedric Martens4:14 PM
PDS_voltage_stable_value = 15
PDS_voltage_rise_max_val = 20
Cedric Martens4:18 PM
http://wiki.ros.org/rospy_tutorials/Tutorials/WritingPublisherSubscriber
modes = stable, rise, fall, not_available
Cedric Martens4:20 PM
Voltage, Currents, Temperatures
Cedric Martens4:24 PM
while()
switch(mode)
stable:
stable(value = 15, noise = 0.05)
rise :
rise(max_value = 20, rate = 0.1V/s, noise = 0.1) 
Cedric Martens4:25 PM
interface PDS_element
def apply(args):
VoltagePDSElement implements PDS_element
TemperaturePDSElement implements PDS_element
voltage = new VoltageElement()
Temperatures[2]= (spawn 3 temps objects0
list.(add voltage and temperatures)
while()
Cedric Martens4:27 PM
for each PDS element
apply
Cedric Martens4:29 PM
while()
for each element:
switch(element.mode)
stable:
element.stable(value = 15, noise = 0.05)
rise :
element.rise(max_value = 20, rate = 0.1V/s, noise = 0.1)

**Comment by MartensCedric** [comment]
Progress on this PR? @adrientremblay :eye: :lips: :eye: :ear: ?

**Comment by MartensCedric** [comment]
adrien from telegram : "Adrien Tremblay, [30.05.20 11:29]
Lmao I'll get to that.  I'm just been busy.  I got docker basically working.  Just some bullshit with setting up the user so that the bash aliases work"

**Comment by MartensCedric** [comment]
@ginywiny will reopen once ready :sunglasses:

**Pull request closed without merging**